packages/manager/apps/pci-block-storage/src/pages/new/components/VolumeTypeStep.component.tsx

```diff
@@ -49,84 +48,32 @@ export function VolumeTypeStep({
   onSubmit,
 }: Readonly<VolumeTypeStepProps>) {
   const { t } = useTranslation(['stepper', 'add', 'common']);
-  const { formatBytes } = useBytes();
-  const { getFormattedCatalogPrice } = useCatalogPrice(6, {
-    hideTaxLabel: true,
-  });
   const { data } = useVolumeModels(projectId, region.name);
   const { defaultValue: encryptionDefaultValue } = useVolumeEncryptions();
   const { ovhSubsidiary } = useContext(ShellContext).environment.getUser();
 
-  const volumeTypes = useMemo(
-    () =>
-      data?.map((m) => ({
-        ...m,
-        label: m.displayName,
-        description:
-          m.availabilityZonesCount !== null
-            ? t(
-                'add:pci_projects_project_storages_blocks_add_type_availability_zone',
-                { count: m.availabilityZonesCount },
-              )
-            : undefined,
-        badges: [
-          m.encrypted
-            ? {
-                label: t(
-                  'common:pci_projects_project_storages_blocks_encryption_available',
-                ),
-                backgroundColor: '#D2F2C2',
-                textColor: '#113300',
-                icon: 'lock' as const,
-              }
-            : {
-                label: t(
-                  'common:pci_projects_project_storages_blocks_encryption_unavailable',
-                ),
-                backgroundColor: '#FFCCD9',
-                textColor: '#4D000D',
-                icon: 'lock' as const,
-              },
-        ],
-        features: [
-          m.iops,
-          t(
-            'add:pci_projects_project_storages_blocks_add_type_addon_capacity_max',
-            {
-              capacity: formatBytes(m.capacity.max),
-            },
-          ),
-        ].concat(m.bandwidth ? [m.bandwidth] : []),
-        price: m.hourlyPrice,
-      })) || [],
-    [data, region, getFormattedCatalogPrice],
-  );
-  const [volumeType, setVolumeType] = useState<typeof volumeTypes[number]>(
-    undefined,
-  );
+  const [volumeModel, setVolumeModel] = useState<TVolumeModel>(undefined);
   const [encryptionType, setEncryptionType] = useState<EncryptionType | null>(
     null,
   );
 
   useEffect(() => {
-    if (volumeType)
-      setEncryptionType(volumeType.encrypted ? encryptionDefaultValue : null);
-  }, [volumeType]);
+    if (volumeModel)
+      setEncryptionType(volumeModel.encrypted ? encryptionDefaultValue : null);
+  }, [volumeModel]);
 
   const attachGuideLink =
     MULTI_ATTACH_INFO_URL[ovhSubsidiary] || MULTI_ATTACH_INFO_URL.DEFAULT;
 
   return (
     <>
-      <TilesInput
-        name="volume-type"
-        label=""
-        value={volumeType}
-        elements={volumeTypes}
-        onChange={(e) => setVolumeType(e)}
+      <VolumeTypeTilesInput
+        value={volumeModel}
+        data={data}
+        onChange={(e) => setVolumeModel(e)}
```

Comment:
```suggestion
        onChange={setVolumeModel}
```
Also it is not an event so `e` doesn't make sense